src/base.ts

```diff
@@ -23,20 +32,7 @@ export default abstract class extends Command {
     }
   }
 
-  async recordActionExecuted(action: string, metadata: MetricMetadata = {}, rawDocument?: string) {
```

Comment:
If you keep this method as it was, you could just pass the `this.specFile` to it and metadata will be automatically collected.

You just need to ensure you set `this.specFile` in the commands. E..g in the validate command, you can do `this.specFile = await load(filePath);`